oracle/weblogic-kubernetes-operator
# Verify pod securityContext/uid behavior on OKD

Test scenario:
* Create a WebLogic domain using WLST in a persistent volume.
   *
   * Start a WebLogic domain with 
   * (a) JMS File store with custom directory assigned to WLS cluster 
   *   e.g. /shared/domain-ns/domains/domain-uid/JmsFileStores
   * (b) WLDF system resource assigned to WLS cluster
   * (c) JDBC system resource assigned to WLS cluster 
   *
   * Print out UID, GID and SELinux label of pods in the domain namespace
   *
   * Send 100 persistent messages to JMS Destination on managed server(2)
   * Stop/Start the managed server(2) by scaling the cluster
   *
   * Print out UID, GID and SELinux label of pods in the domain namespace
   * 
   * Make sure all 100 persistent messages are recovered form managed server(2)
   
******************************************************* 

Test passed at OKD: https://build.weblogick8s.org:8443/job/wko34-okd/24/
From https://build.weblogick8s.org:8443/job/wko34-okd/24/console, using "got uid" we can see in the domain namespace, all the pods have the same UID, GID and SELinux label at initial domain startup and after managed server 2 was restarted. 

<Initial domain startup, got uid recovery-dpv-admin-server runAsUser: 1000780000 fsGroup: 1000780000 seLinuxOptions: s0:c28,c12 for pod recovery-dpv-admin-server in the namespace ns-xcpbmn>

<Initial domain startup, got uid recovery-dpv-managed-1 runAsUser: 1000780000 fsGroup: 1000780000 seLinuxOptions: s0:c28,c12 for pod recovery-dpv-managed-1 in the namespace ns-xcpbmn>

<Initial domain startup, got uid recovery-dpv-managed-2 runAsUser: 1000780000 fsGroup: 1000780000 seLinuxOptions: s0:c28,c12 for pod recovery-dpv-managed-2 in the namespace ns-xcpbmn>

<After managed Server2 was restarted, got uid recovery-dpv-admin-server runAsUser: 1000780000 fsGroup: 1000780000 seLinuxOptions: s0:c28,c12 for pod recovery-dpv-admin-server in the namespace ns-xcpbmn>

<After managed Server2 was restarted, got uid recovery-dpv-managed-1 runAsUser: 1000780000 fsGroup: 1000780000 seLinuxOptions: s0:c28,c12 for pod recovery-dpv-managed-1 in the namespace ns-xcpbmn>

<After managed Server2 was restarted, got uid recovery-dpv-managed-2 runAsUser: 1000780000 fsGroup: 1000780000 seLinuxOptions: s0:c28,c12 for pod recovery-dpv-managed-2 in the namespace ns-xcpbmn>
